frontend/pages/404.tsx

```diff
@@ -0,0 +1,50 @@
+import { Center, Text, Divider, VStack } from '@chakra-ui/react';
+import { WithProvidersProps } from '@components/common';
+import {
+   DefaultLayout,
+   getLayoutServerSideProps,
+   WithLayoutProps,
+} from '@layouts/default';
+import { GetStaticProps } from 'next';
+
+type Custom404Props = WithProvidersProps<WithLayoutProps<{}>>;
+
+export const Custom404: NextPageWithLayout<Custom404Props> = () => {
+   return (
+      <Center flexGrow={1}>
+         <VStack>
+            <Text fontSize={36} fontWeight="bold">
+               404
+            </Text>
+            <Text fontSize={24} fontWeight="semibold" paddingBottom="24px">
+               Page not Found
+            </Text>
+            <Divider borderWidth="1px" borderColor="red.500" width="50px" />
+            <Text
+               fontSize={20}
+               paddingTop="24px"
+               overflowWrap="break-word"
+               align="center"
+               maxW="360px"
+            >
+               We can&#39;t find the page you&#39;re looking for. Whoops.
+            </Text>
+         </VStack>
+      </Center>
+   );
+};
+
+Custom404.getLayout = function getLayout(page, pageProps) {
+   return <DefaultLayout {...pageProps.layoutProps}>{page}</DefaultLayout>;
+};
+
+export const getStaticProps: GetStaticProps<Custom404Props> = async () => {
+   return {
+      props: {
+         layoutProps: await getLayoutServerSideProps(),
+         appProps: {},
+      },
```

Comment:
Does this come from an example or something of that sort? I'm not familiar with `layoutProps` and `appProps`, and I'd be interested to learn more.

Reply:
These are props that are common across pages. For example, the product page props:

https://github.com/iFixit/react-commerce/blob/34d0a7010f673a10ab394b46355c4c6a3eb75bf2/frontend/templates/product/hooks/useProductTemplateProps.ts#L6-L10

The props for the 404 component have the type:

https://github.com/iFixit/react-commerce/blob/3e20e8bdeadaecfc341588bb4310d63af4907e20/frontend/pages/404.tsx#L10

Which looks like:

https://github.com/iFixit/react-commerce/blob/d92212079722972e8a53ef5c42e1c0d4a388236c/frontend/components/common/AppProviders.tsx#L23

https://github.com/iFixit/react-commerce/blob/56d32edbfe6a4db025f266e6328dd515f202d1d6/frontend/layouts/default/types.ts#L10

I wonder if we can drop `WithProvidersProps`, since we're not using Algolia on this page. Wanna give it a try @rickisXP?

The layout props are necessary for getting default layout styles, like which store to use in the header for example.

Reply:
@rickisXP Do we still need to pass `appProps` now that we're no longer using `WithProviderProps`?